test/pipelines/test_ray.py

```diff
@@ -14,21 +14,25 @@ def shutdown_ray():
     try:
         import ray
 
+        ray.serve.shutdown()
         ray.shutdown()
     except:
         pass
 
 
 @pytest.mark.integration
 @pytest.mark.parametrize("document_store_with_docs", ["elasticsearch"], indirect=True)
-def test_load_pipeline(document_store_with_docs):
+@pytest.mark.parametrize("serve_detached", [True, False])
+def test_load_pipeline(document_store_with_docs, serve_detached):
     pipeline = RayPipeline.load_from_yaml(
         SAMPLES_PATH / "pipeline" / "ray.haystack-pipeline.yml",
         pipeline_name="ray_query_pipeline",
         ray_args={"num_cpus": 8},
+        serve_args={"detached": serve_detached},
     )
     prediction = pipeline.run(query="Who lives in Berlin?", params={"Retriever": {"top_k": 10}, "Reader": {"top_k": 3}})
 
+    assert pipeline._serve_controller_client._detached == serve_detached
```

Comment:
Amazing, thank you! That looks very private, I hope it won't change often :smile: Good for testing though :+1: